addons/survey/static/src/question_page/question_page_list_renderer.js

```diff
@@ -51,28 +51,27 @@ export class QuestionPageListRenderer extends ListRenderer {
     }
 
     getSectionColumns(columns) {
-        const sectionColumns = [];
-
         let titleColumnIndex = 0;
+        let found = false;
+        let colspan = 1
         for (const col of columns) {
-            if (col.type !== "field") {
+            if (!found && col.name !== this.titleField) {
                 continue;
             }
-            if (col.widget === "handle") {
-                titleColumnIndex = 1;
-                sectionColumns.push(col);
+            if (!found) {
+                found = true;
+                titleColumnIndex = columns.indexOf(col);
```

Comment:
We can avoid looking up the list here, since we are already looping it: either
* Loop over indices in range of `columns` length (with or without reassigning `col=columns[idx]` for readability), *or*
* Use `titleColumnIndex++;` (rename to `potentialTitleColumIndex` ?) on line 59 and `titleColumnIndex = found ? titleColumnIndex : 0;` after the loop